### DISCUSSION TITLE
Permission denied: '/root/.art'

### DISCUSSION THREAD

**Discussion by keode** [Trusted-AI/adversarial-robustness-toolbox#1795] [Q&A]
I installed art in a conda virtual environment via pip install adversarial-robustness-toolbox and the installation went through without error. I then tried to run the example get_started_scikit_learn.py, but I got the error

> (art) ray@T480s:~/adversarial-robustness-toolbox$ python examples/get_started_scikit_learn.py 
> Traceback (most recent call last):
>   File "/home/ray/adversarial-robustness-toolbox/examples/get_started_scikit_learn.py", line 10, in <module>
>     from art.attacks.evasion import FastGradientMethod
>   File "/home/ray/anaconda3/envs/art/lib/python3.10/site-packages/art/__init__.py", line 7, in <module>
>     from art import attacks
>   File "/home/ray/anaconda3/envs/art/lib/python3.10/site-packages/art/attacks/__init__.py", line 8, in <module>
>     from art.attacks import evasion
>   File "/home/ray/anaconda3/envs/art/lib/python3.10/site-packages/art/attacks/evasion/__init__.py", line 4, in <module>
>     from art.attacks.evasion.adversarial_patch.adversarial_patch import AdversarialPatch
>   File "/home/ray/anaconda3/envs/art/lib/python3.10/site-packages/art/attacks/evasion/adversarial_patch/adversarial_patch.py", line 31, in <module>
>     from art.attacks.evasion.adversarial_patch.adversarial_patch_numpy import AdversarialPatchNumpy
>   File "/home/ray/anaconda3/envs/art/lib/python3.10/site-packages/art/attacks/evasion/adversarial_patch/adversarial_patch_numpy.py", line 37, in <module>
>     from art.estimators.estimator import BaseEstimator, NeuralNetworkMixin
>   File "/home/ray/anaconda3/envs/art/lib/python3.10/site-packages/art/estimators/__init__.py", line 4, in <module>
>     from art.estimators.estimator import (
>   File "/home/ray/anaconda3/envs/art/lib/python3.10/site-packages/art/estimators/estimator.py", line 27, in <module>
>     from art.config import ART_NUMPY_DTYPE
>   File "/home/ray/anaconda3/envs/art/lib/python3.10/site-packages/art/config.py", line 94, in <module>
>     set_data_path(_config["ART_DATA_PATH"])
>   File "/home/ray/anaconda3/envs/art/lib/python3.10/site-packages/art/config.py", line 47, in set_data_path
>     os.makedirs(expanded_path, exist_ok=True)
>   File "/home/ray/anaconda3/envs/art/lib/python3.10/os.py", line 215, in makedirs
>     makedirs(head, exist_ok=exist_ok)
>   File "/home/ray/anaconda3/envs/art/lib/python3.10/os.py", line 225, in makedirs
>     mkdir(name, mode)
> PermissionError: [Errno 13] Permission denied: '/root/.art'
> 

It seems that _config["ART_DATA_PATH"] is set to a root directory. Why is that so and how can I solve this?

My installation commands:

> 
> conda create --name art
> conda activate art
> pip install adversarial-robustness-toolbox

My pip list output

> (art) ray@T480s:~/adversarial-robustness-toolbox$ pip list
> Package                        Version
> ------------------------------ ---------
> adversarial-robustness-toolbox 1.11.0
> certifi                        2022.6.15
> joblib                         1.1.0
> llvmlite                       0.39.0
> numba                          0.56.0
> numpy                          1.22.4
> pip                            22.1.2
> scikit-learn                   1.0.2
> scipy                          1.8.1
> setuptools                     61.2.0
> six                            1.16.0
> threadpoolctl                  3.1.0
> tqdm                           4.64.0
> wheel                          0.37.1
>

**Comment by beat-buesser** [comment]
Hi @keode Did you install ART as root?

**Reply by keode** [reply]
Hi @beat-buesser Thanks for your prompt reply. No. I did not use sudo pip to install art.. I used the following commands to install it (in a conda virtual environment though)
`conda create --name art` #create a conda virtual environment
`conda activate art` #so this command activates the virtual environment and I see the environment name (art) preceding every command prompt. Then I install art by
`pip install adversarial-robustness-toolbox`

**Reply by beat-buesser** [reply]
@keode Ok, I see. I was asking about root because the folder `.art` is by default created in the user's home directory. What is the content of your environment variable `HOME`?
